cilium/bpf_metadata.cc

```diff
@@ -167,22 +167,35 @@ bool Config::getMetadata(Network::ConnectionSocket& socket) {
     dip = socket.localAddress()->ip();
   }
 
+  // Sidecars may get incorrect "ingress" config, override.
```

Comment:
We don't seem to be validating the read value at L174 before it's overridden?

Reply:
Also, I'm not able to comment on the lines that are not part of the change.
But are L154-163 still valid?